web/src/main.css

```diff
@@ -0,0 +1,6 @@
+.inline-list-item {
+  display: inline;
+}
+.inline-list-item:not(:last-child)::after {
+  content: ', ';
+}
```

Comment:
This can be defined as a shortcut:

```ts
shortcuts: {
  'inline-list-item': 'inline not-last:after:content-[",_"]'
}
```

Reply:
I suppose, but I'm also reluctant to add *more* Tailwindisms at this point (#54).

Reply:
Converted it to a CSS module: b04dab6fab7bc49ecab449d3d0e83059fab9d744

Reply:
Okay what are the pain points when using Uno? I guess we can use individual files for nice stuff, but Uno is pretty good at covering 95% of the styles we need.

Reply:
It's "solving" a non-problem. It's adding an extra step of indirection to all docs lookups (instead of `mdn -> code` (or just writing the code directly) everything turns into `mdn -> docs.uno -> code -> docs.uno to understand wtf is going on`) for no benefit. `class` quickly turns into an unmanageable mess, especially since you can't really do comments (without factoring it out of the immediate `class`). The Firefox debugger isn't aware of it, so I still end up writing it as real CSS first *anyway* before then converting it into Tailwind gibberish, so the idea of it even saving any keystrokes in the end is absurd. And it pushes you away from CSS features like grids that can help make your stylesheets significantly more understandable, since they can't really map up to static class sets.

And all of this, just so that you can write `ml-1` instead of `margin-left: 0.25rem;`.

I'm not against abstraction. I'm not against the idea of CSS frameworks. But they have to provide actual value to make up for the costs involved when using them. Tailwind provides none of that. It's just [ProGuard](https://www.guardsquare.com/proguard) for people who thought that that was too automated.

Reply:
> It's adding an extra step of indirection to all docs lookups (instead of mdn -> code (or just writing the code directly) everything turns into mdn -> docs.uno -> code -> docs.uno to understand wtf is going on) for no benefit.

This is not an issue for me personally. I know what I need/want to use 95% of the time. Using these classes *has* a learning curve, but it isn't super steep in my opinion.

> `class` quickly turns into an unmanageable mess, especially since you can't really do comments (without factoring it out of the immediate `class`). 

Agreed, it can messy pretty quick. This is were shortcuts and separation into components helps tho. Plain CSS files can also get super messy (and long).

> And it pushes you away from CSS features like grids that can help make your stylesheets significantly more understandable

How? You can still use all the new fancy layout stuff, like grid, flex, etc...

> I'm not against abstraction. I'm not against the idea of CSS frameworks. But they have to provide actual value to make up for the costs involved when using them. Tailwind provides none of that.

Agreed, and I think Uno *provides* advantages and makes your life easier when writing CSS. I find it quite easy to use them and writing CSS is more costly for me personally (which I did many years, ugh).

> It's just [ProGuard](https://www.guardsquare.com/proguard) for people who thought that that was too automated.

Yes ProGuard is garbage :)

Reply:
> This is not an issue for me personally. I know what I need/want to use 95% of the time. Using these classes has a learning curve, but it isn't super steep in my opinion.

And, again, what do you gain in exchange for this extra obfuscation?

> Agreed, it can get messy pretty quick. This is where shortcuts and separation into components helps tho. Plain CSS files can also get super messy (and long).

And that's an improvement over plain old CSS.. how?

> How? You can still use all the new fancy layout stuff, like grid, flex, etc...

https://unocss.dev/interactive/?s=grid-template

Reply:
> And, again, what do you gain in exchange for this extra obfuscation?

Maintainability, unified CSS code style, unified access for sizing, colors, spacing, no unused CSS rules, etc...

> And that's an improvement over plain old CSS.. how?

No nesting, obnoxious variable declaration and usage, possible multiple implementations of the same rules, hundreds or even thousands of lines of code across many files.

> https://unocss.dev/interactive/?s=grid-template

You can still use `grid-cols-<num>` and `grid-rows-<num>`. Tbh, I never needed nor used grid areas.

Reply:
> no unused CSS rules

[I've got a bridge to sell you](https://unpkg.com/tailwindcss@2.2.19/dist/tailwind.css)

Reply:
All these rules won't even be generated. The current generated CSS is only 110 rules with a size of 7.5 kB (not minified).

Reply:
The rules are *effectively* there and need to be taken into account. (Of course, this is only a tiny fraction of the infinite number of rules that actually exist once you take Uno's dynamic rules into account.)

Reply:
I really don't get what you are trying to tell me here. All generated CSS is based on the classes you _actually_ use, not those which _could_ be there if you used all possible class combinations. Why should we care about **ALL** possible rules, if we don't use them and thus will not be generated?

If I manually maintain CSS rules, I can still just re-declare the same set of rules using different selectors. I could write the same block of rules 10 times and each time I use a different selector both in the CSS and the HTML (TSX) file. This bloat basically is only discovered if you go over the CSS with a specialized tool or manually look for these kind of issues and fix them yourself.

Reply:
> Why should we care about ALL possible rules, if we don't use them and thus will not be generated?

We still have to act as if the names exist and are reserved by the framework.

> If I manually maintain CSS rules, I can still just re-declare the same set of rules using different selectors. I could write the same block of rules 10 times and each time I use a different selector both in the CSS and the HTML (TSX) file. This bloat basically is only discovered if you go over the CSS with a specialized tool or manually look for these kind of issues and fix them yourself.

And you can declare identical TSX components ten times over too, I don't see how that's an argument one way or another.

Besides, I'd argue that having identical rules isn't a problem as long as they have separate semantic intents.

Reply:
Resolved, we use SCSS in the future.